x-pack/metricbeat/module/aws/cloudwatch/metadata/ec2/ec2.go

```diff
@@ -33,25 +34,35 @@ func AddMetadata(logger *logp.Logger, regionName string, awsConfig awssdk.Config
 		return events, fmt.Errorf("aws.ec2.instance fields are not available, skipping region %s: %w", regionName, err)
 	}
 
-	// collect monitoring state for each instance
-	monitoringStates := map[string]string{}
-	for instanceID, output := range instancesOutputs {
-		for eventIdentifier := range events {
-			eventIdentifierComponents := strings.Split(eventIdentifier, "-")
-			potentialInstanceID := strings.Join(eventIdentifierComponents[0:len(eventIdentifierComponents)-1], "-")
+	for eventIdentifier := range events {
+		eventIdentifierComponents := strings.Split(eventIdentifier, "-")
+		potentialInstanceID := strings.Join(eventIdentifierComponents[0:len(eventIdentifierComponents)-1], "-")
+
+		// add host cpu/network/disk fields and host.id and rate metrics for all instances from both the monitoring
+		// account and linked source accounts if include_linked_accounts is set to true
+		addHostFields(events[eventIdentifier], potentialInstanceID)
+		period, err := events[eventIdentifier].RootFields.GetValue(aws.CloudWatchPeriodName)
+		if err != nil {
+			logger.Warnf("can't get period information for instance %s, skipping rate calculation", eventIdentifier)
+		} else {
+			calculateRate(events[eventIdentifier], period.(int))
+		}
+
+		// add instance ID from dimension value
+		dimInstanceID, _ := events[eventIdentifier].RootFields.GetValue("aws.dimensions.InstanceId")
```

Comment:
does this always exist in the dimensions? should we check the error here just in case?

Reply:
Yeah good point! Will add it!

Reply:
I did some testing on this, when `aws.dimensions.InstanceId` doesn't exist, `dimInstanceID` returns as nil. Then `events[eventIdentifier].RootFields.Put("cloud.instance.id", dimInstanceID)` will just put nil as the value for `cloud.instance.id` which won't show up in the event.

Reply:
😬 that wouldn't be ideal - i wonder if we could fall back to the instance ID we from the loop var?

Reply:
I added the error check here to make sure only put value to `cloud.instance.id` when `aws.dimensions.InstanceId` exists. We can't use instance ID from the inner for loop var because that for loop is going through all the instances metadata from `DescribeInstances` API. We are trying to use the two for loops to match events with the dimension InstanceId to the same ID from `DescribeInstances`.